core/src/main/java/org/testcontainers/images/AbstractImagePullPolicy.java

```diff
@@ -0,0 +1,42 @@
+package org.testcontainers.images;
+
+import lombok.extern.slf4j.Slf4j;
+import org.slf4j.Logger;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.DockerLoggerFactory;
+
+@Slf4j
+public abstract class AbstractImagePullPolicy implements ImagePullPolicy {
+
+    private static final LocalImagesCache LOCAL_IMAGES_CACHE = LocalImagesCache.INSTANCE;
+
+    @Override
+    public boolean shouldPull(DockerImageName imageName) {
+        Logger logger = DockerLoggerFactory.getLogger(imageName.toString());
+
+        // Does our cache already know the image?
+        ImageData imageData = LOCAL_IMAGES_CACHE.get(imageName);
+        if (imageData != null) {
+            logger.trace("{} is already in image name cache", imageName);
+        } else {
+            logger.debug("{} is not in image name cache, updating...", imageName);
+            // Was not in cache, inspecting...
+            imageData = LOCAL_IMAGES_CACHE.refreshCache(imageName).orElse(null);
+        }
+
+        if (imageData != null && !shouldPullCached(imageName, imageData)) {
+            return false;
+        }
+
+        log.trace("Should pull image: {}", imageName);
```

Comment:
I have the feeling that we should always log, maybe at debug level, about the pull decision (including if we've decided not to pull!)